src/celeritas/io/EventReader.hh

```diff
@@ -50,23 +50,31 @@ class EventReader : public EventReaderInterface
 
   public:
     // Construct from a filename
-    EventReader(std::string const& filename, SPConstParticles params);
+    EventReader(std::string const& filename, SPConstParticles particles);
 
     //! Prevent copying and moving
     CELER_DELETE_COPY_MOVE(EventReader);
 
     // Read a single event from the event record
     result_type operator()() final;
 
+    //! Get total number of events
+    size_type num_events() const { return num_events_; }
```

Comment:
Since this function is being added to `RootEventReader` too, I think you can go ahead and make it part of the `EventReaderInterface`?